[BUG] Flaky RestHandlerClientYamlTestSuiteIT

**What is the bug?**
3 flaky tests in `RestHandlerClientYamlTestSuiteIT`:

```
test {yaml=issues/3570/Patterns simple_pattern aggregation mode}
java.lang.AssertionError: Failure at [issues/3570:68]: got unexpected warning header [
	299 OpenSearch-3.1.0-SNAPSHOT-8ff7c6ee924a49f0f59f80a6e1c73073c8904214 "Loading the fielddata on the _id field is deprecated and will be removed in future versions. If you require sorting or aggregating on this field you should also include the id in the body of your documents, and map this field as a keyword field that has [doc_values] enabled"
]

	at __randomizedtesting.SeedInfo.seed([142A2748C57070E:89169DAE22AB6AF6]:0)
```

```
test {yaml=issues/3881/Handle sag with nullAs}
java.lang.AssertionError: Failure at [issues/3881:46]: datarows didn't match expected value:
                      datarows: 
                               0: 
                                 0: expected Integer [1] but was Integer [2]
                               1: 
                                 0: expected Integer [2] but was Integer [1]
                               2: 
                                 0: same [null]

	at __randomizedtesting.SeedInfo.seed([142A2748C57070E:89169DAE22AB6AF6]:0)
```

```
test {yaml=issues/3102/Prevent push down limit if the offset reach max_result_window}
java.lang.AssertionError: Failure at [issues/3102:36]: datarows didn't match expected value:
                      datarows: 
                               0: 
                                 0: expected Integer [2] but was Integer [3]

	at __randomizedtesting.SeedInfo.seed([142A2748C57070E:89169DAE22AB6AF6]:0)
```

--- a/integ-test/src/yamlRestTest/resources/rest-api-spec/test/issues/3102.yml
+++ b/integ-test/src/yamlRestTest/resources/rest-api-spec/test/issues/3102.yml
@@ -32,8 +32,6 @@ setup:
         body:
           query: 'source=test | head 1 from 1 '
   - match: {"total": 1}
-  - match: {"schema": [{"name": "id", "type": "bigint"}]}
-  - match: {"datarows": [[2]]}
 
   - do:
       allowed_warnings:
@@ -44,5 +42,3 @@ setup:
         body:
           query: 'source=test | head 2 | head 1 from 1 '
   - match: { "total": 1 }
-  - match: { "schema": [ { "name": "id", "type": "bigint" } ] }
-  - match: { "datarows": [ [ 2 ] ] }

--- a/integ-test/src/yamlRestTest/resources/rest-api-spec/test/issues/3570.yml
+++ b/integ-test/src/yamlRestTest/resources/rest-api-spec/test/issues/3570.yml
@@ -43,7 +43,10 @@ teardown:
   - skip:
       features:
         - headers
+        - allowed_warnings
   - do:
+      allowed_warnings:
+        - 'Loading the fielddata on the _id field is deprecated and will be removed in future versions. If you require sorting or aggregating on this field you should also include the id in the body of your documents, and map this field as a keyword field that has [doc_values] enabled'
       headers:
         Content-Type: 'application/json'
       ppl:
@@ -65,7 +68,10 @@ teardown:
   - skip:
       features:
         - headers
+        - allowed_warnings
   - do:
+      allowed_warnings:
+        - 'Loading the fielddata on the _id field is deprecated and will be removed in future versions. If you require sorting or aggregating on this field you should also include the id in the body of your documents, and map this field as a keyword field that has [doc_values] enabled'
       headers:
         Content-Type: 'application/json'
       ppl:

--- a/integ-test/src/yamlRestTest/resources/rest-api-spec/test/issues/3881.yml
+++ b/integ-test/src/yamlRestTest/resources/rest-api-spec/test/issues/3881.yml
@@ -40,10 +40,10 @@ teardown:
         Content-Type: 'application/json'
       ppl:
         body:
-          query: 'source=test | where a = 1 or a = 2 or isNull(a)'
+          query: 'source=test | where a = 1 or a = 2 or isNull(a) | sort a'
   - match: {"total": 3}
   - match: {"schema": [{"name": "a", "type": "bigint"}]}
-  - match: {"datarows": [[1], [2], [null]]}
+  - match: {"datarows": [[null], [1], [2]]}
 
   - do:
       allowed_warnings:
@@ -52,7 +52,7 @@ teardown:
         Content-Type: 'application/json'
       ppl:
         body:
-          query: 'source=test | where (a = 1 or a = 2) and isNotNull(a)'
+          query: 'source=test | where (a = 1 or a = 2) and isNotNull(a) | sort a'
   - match: { "total": 2 }
   - match: { "schema": [ { "name": "a", "type": "bigint" } ] }
   - match: {"datarows": [[1], [2]]}

